### DISCUSSION TITLE
How does the OSC52 change work?

### DISCUSSION THREAD

**Discussion by surmish** [LazyVim/LazyVim#2715] [Q&A] [answered]
Regarding [this](https://github.com/LazyVim/LazyVim/commit/21ca466db13e68740aa3348c7406369c53062a9e) change, I tried yanking some text into the `+` register using `"+y` keys, but the yanked text does not show up outside of the neovim sessions. I'm using `wezterm` on windows to ssh into remote linux machines.

The neovim specific plugin (https://github.com/ojroques/nvim-osc52) also doesn't seem to work for me. What did work, was the old vim plugin (https://github.com/ojroques/vim-oscyank). Any reason why this might be the case?

**Comment by folke** [comment]
The Neovim version in your ssh instance needs to be nightly in order for this to work

**Reply by folke** [reply]
wait, I see why it won't work .
Just a sec :)

**Comment by folke** [comment, marked as answer]
Seems like the feature was changed in the meantime.
I just pushed an update https://github.com/LazyVim/LazyVim/commit/1c9f4160a22dbdca2db36169460fcf28641607e7

Check `:h clipboard.osc52` for more details.

**Comment by james-m-henderson** [comment]
I can say that for me the solution was to use `unnamedplus` as my clipboard, even in the remote session.

So in options.lua:
```
vim.opt.clipboard = "unnamedplus"

vim.g.clipboard = {
  name = "OSC 52",
  copy = {
    ["+"] = require("vim.ui.clipboard.osc52").copy("+"),
    ["*"] = require("vim.ui.clipboard.osc52").copy("*"),
  },
  paste = {
    ["+"] = require("vim.ui.clipboard.osc52").paste("+"),
    ["*"] = require("vim.ui.clipboard.osc52").paste("*"),
  },
}
```

In kitty conf file (`~/.config/kitty/kitty.conf`):
```
clipboard_control write-clipboard write-primary read-clipboard read-primary
```
It seems that nvim will want to read the clipboard quite a bit and kitty considers this a security vulnerability so it will ask first. This disables that.

This is using kitty as the terminal sshing from darwin (macos) to linux, and no special settings otherwise.

In addition the following can be used to test outside of vim if your OSC 52 codes are working OK regardless of terminal type or vim version etc.

```
printf $'\e]52;c;%s\a' "$(base64 <<<'copy this')"
```

If that appears in your system paste buffer, then the issue is with vim config not your terminal.

Version info:
```
NVIM v0.10.3
Build type: Release
LuaJIT 2.1.1713484068

lazyvim:
        version 14.7.0
        tag     v14.7.0
        branch  main
        commit  3660c52
```

Also LazyVim will show you the registers if you type in `"` in command mode, you should see OSC 52 for * and + if the terminal is working properly.

Hopefully this speeds up somebody else's debugging process.

**Reply by kickymaulana** [reply]
thanks this work

**Reply by tvlpirb** [reply]
I can confirm that this fixed the issues for me.

**Reply by michaelfortunato** [reply]
This is the only thing that worked. Thank you.

**Reply by michaelfortunato** [reply]
Do you know if its possible to set clipboard control based on the host?

**Reply by james-m-henderson** [reply]
According to a reddit thread I saw it is like this:

```
if vim.fn.hostname() == "blah blah" then 
  -- do thing
end
```

https://www.reddit.com/r/neovim/comments/w90lp2/if_hostname_in_lua_and_some_other_conversions/

**Reply by ripo4ek** [reply]
man, i was fighting with this issue the whole day, thx!

**Reply by nguyenvulong** [reply]
this works for WezTerm as well, thank you!
